src/app/util/ember-compatibility-functions.cpp

```diff
@@ -199,6 +199,23 @@ CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter * ap
                   ChipLogValueMEI(aClusterInfo.mClusterId), ChipLogValueX64(aClusterInfo.mNodeId), aClusterInfo.mEndpointId,
                   aClusterInfo.mFieldId, aClusterInfo.mListIndex);
 
+    AttributeAccessInterceptor * interceptor = findAttributeAccessInterceptor(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+    if (interceptor != nullptr)
+    {
+        bool dataRead;
+        // TODO: We should probably clone the writer and convert failures here
+        // into status responses, unless our caller already does that.
+        ReturnErrorOnFailure(interceptor->ReadAttributeData(aClusterInfo, apWriter, &dataRead));
+
+        if (dataRead)
+        {
+            // TODO: Add DataVersion support
+            ReturnErrorOnFailure(
+                apWriter->Put(chip::TLV::ContextTag(AttributeDataElement::kCsTag_DataVersion), kTemporaryDataVersion));
+            return CHIP_NO_ERROR;
+        }
+    }
+
     EmberAfAttributeType attributeType;
```

Comment:
Shouldn't there be some kind of conditional check here to only continue ahead with the usual `EmberAfReadAttribute` call if we didn't successfully read out an attribute from a registered accessor?

Reply:
Yes, that check is the `if (dataRead)` check.  Or am I missing something?